localstack/aws/serving/asgi.py

```diff
@@ -39,14 +39,14 @@ def __init__(
         self,
         gateway: Gateway,
         event_loop: Optional[AbstractEventLoop] = None,
-        threads: int = 1000,
+        threads: int = None,
         lifespan_listener: Optional[ASGILifespanListener] = None,
         websocket_listener=None,
     ) -> None:
         self.gateway = gateway
 
         self.event_loop = event_loop or asyncio.get_event_loop()
-        self.executor = _ThreadPool(threads, thread_name_prefix="asgi_gw")
+        self.executor = _ThreadPool(threads or 1000, thread_name_prefix="asgi_gw")
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (code_refinement):** Use of `or` for default parameter handling is clear but consider centralizing default values.

While the use of `or` for providing a default value is straightforward, centralizing default values (e.g., as constants) can make future adjustments and readability better.

```suggestion
        threads: int = os.cpu_count() or 4,
```